### DISCUSSION TITLE
Truncating search

### DISCUSSION THREAD

**Discussion by ghost** [thuliteio/doks#1012] [Q&A]
Hey,
I wanted to achieve the effect mentioned in #801. 

In index.js I changed the code around the line 170.
```js
const description = document.createElement('span');
if (doc.description.length > 64) {
  resultIndex = doc.description.indexOf(searchQuery);
  description.textContent = doc.description.substring(resultIndex-32, resultIndex+32)+"...read more";    
} else {                                                           
  description.textContent = doc.description;                     
}
description.textContent = doc.description;
```
I tried deleting cache and running `npm run clean`. I am not sure why but I am not seeing any change.

Another thing to note - but this is maybe problem with my setup or Hugo that in this file I have errors on Hugo syntax:
For example:
![image](https://user-images.githubusercontent.com/49745287/220656971-9ae4949f-8d5d-45bd-8bbc-13bcbdbb8546.png)

Any ideas why this is happening?

**Comment by h-enk** [comment]
It looks like (not tested) you need to drop the last line (`description.textContent = doc.description;`) and probably update `description.textContent = doc.description.substring(resultIndex-32, resultIndex+32)+"...read more";`  to use `64` in stead of `32`:

```js
const description = document.createElement('span');
if (doc.description.length > 64) {
  resultIndex = doc.description.indexOf(searchQuery);
  description.textContent = doc.description.substring(resultIndex-64, resultIndex+64)+"...read more";    
} else {                                                           
  description.textContent = doc.description;                     
}
```

BTW, in the upcoming Doks 1.0 release I'm using `summary` instead of `content` for the search results.

> Another thing to note - but this is maybe problem with my setup or Hugo that in this file I have errors on Hugo syntax:

Nothing to do with your setup. It's because (currently) `index.js` builds the search index using Hugo template code. So, `index.js` invalidates as a JavaScript file. Not nice, I admit. So, that's why I set up things differently/smarter in the upcoming Doks 1.0 release.
